packages/adblocker/src/engine/engine.ts

```diff
@@ -1024,6 +1024,169 @@ export default class FilterEngine extends EventEmitter<EngineEventHandlers> {
       };
     }
 
+    const { matches, allowGenericHides } = this.matchCosmeticFilters({
+      url,
+      hostname,
+      domain,
+      classes,
+      hrefs,
+      ids,
+      getRulesFromDOM,
+      getRulesFromHostname,
+      getInjectionRules,
+      getExtendedRules,
+      getPureHasRules: injectPureHasSafely,
+      callerContext,
+    });
+
+    const filters = [];
+
+    for (const { filter, exception } of matches) {
+      if (exception === undefined) {
+        filters.push(filter);
+      }
+    }
+
+    const { extended, scripts, styles } = this.injectCosmeticFilters(filters, {
+      url,
+      injectScriptlets: getInjectionRules,
+      injectExtended: getExtendedRules,
+      injectPureHasSafely,
+      allowGenericHides,
+      getBaseRules,
+      hidingStyle,
+    });
+
+    return {
+      active: true,
+      extended,
+      scripts,
+      styles,
+    };
+  }
+
+  public injectCosmeticFilters(
```

Comment:
This function doesn't actually inject something to DOM or any other contexts. The name is confusing here, how about: `compileCosmeticsFilters`?

Reply:
adblocker by itself does not inject anything as it is platform agnostic - same thing with the network layer.

I've chosen `inject` as a name as it build on previous nomenclature of the emitted events `style-injected` and `script-injected`. But I don't a strong opinion. 

Alternative names can also be prepare or assemble? @philipp-classen @remusao @seia-soto  what do you think?